### PR TITLE
21866 - Flag-Expired-NR

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* https://github.com/bcgov/entity/issues/21866

*Description of changes:*
- Removed the isIA condition in AffiliatedEntityTable.vue to allow all application types show Flag NR expried
- Run lint so some formatting updated

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 1]
```diff
@@ -176,7 +176,6 @@
 import {
   AffiliationInvitationStatus,
   AffiliationInvitationType,
-  AffiliationTypes,
```
Comment (ArwenQin) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:179]: Unused, removed

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 2]
```diff
@@ -225,18 +224,14 @@ export default defineComponent({
       return NrState.DRAFT === state.toUpperCase()
     }
 
-    const isIA = (type: string): boolean => {
-      return (type === AffiliationTypes.INCORPORATION_APPLICATION || type === AffiliationTypes.REGISTRATION)
-    }
-
```
Comment (ArwenQin) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:231]: Unused, removed

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 3]
```diff
@@ -225,18 +224,14 @@ export default defineComponent({
...
     const isProcessing = (state: string): boolean => {
       return NrDisplayStates.PROCESSING === state
     }
 
-    /** Draft IA with Expired NR */
+    /** Draft Application with Expired NR */
     const isExpired = (item: Business): boolean => {
       return isDraft(status(item)) && (item.nameRequest && (item.nameRequest.expirationDate !== null) &&
-        (new Date(item.nameRequest.expirationDate) < new Date())) && isIA(type(item))
```
Comment (ArwenQin) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:239]: Removed this condition to allow Flag for all types

**Comment by ArwenQin** [comment]
Now showing the flag:
image

image

**Review by seeker25 — APPROVED** [review]

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 4]
```diff
@@ -225,18 +224,14 @@ export default defineComponent({
...
+        (new Date(item.nameRequest.expirationDate) < new Date()))
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:234]: Not your code but:

 "parsing dates via the Date object is unreliable and inconsistent across platforms"
"There are different implementations and different behaviors. Sometimes you may get local time. Sometimes you may get UTC. Sometimes you may get Invalid Date. Sometimes you will get different results near DST transitions."

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:234]: better using `moment(item.nameRequest.expirationDate)` or similar

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 2 above)
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:231]: On line 228, you can write that `type` is "AffiliationTypes".

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 4 above)
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:234]: What are you thoughts on these?

- https://sumitaec108.medium.com/stop-using-momentjs-f71872f01716
- https://dockyard.com/blog/2020/02/14/you-probably-don-t-need-moment-js-anymore
- https://www.reddit.com/r/javascript/comments/b0ky8z/why_you_shouldnt_use_momentjs/
- https://bootcamp.uxdesign.cc/are-you-still-using-moment-js-in-your-new-project-try-these-four-libraries-instead-8ea84cd50541

**Pull request merged**